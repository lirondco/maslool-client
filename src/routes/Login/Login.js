import React, { Component } from 'react'
import LoginForm from '../../components/LoginForm/LoginForm'
import './Login.css'

export default class Login extends Component{
    static defaultProps = {
        location: {},
        history: {
            push: () => {}
        },
    }

    handleLoginSuccess = () => {
        const { location, history } = this.props
        const destination = (location.state || {}).from || '/trails'
        history.push(destination)
    }

    render() {
        return (
            <section className = 'loginpage'>
                <LoginForm 
                    onLoginSuccess={this.handleLoginSuccess}
                />
            </section>
        )
    }
}