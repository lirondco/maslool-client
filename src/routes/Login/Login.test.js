import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Login from './Login'

describe(`Login route component`, () => {
  it('renders the complete page', () => {
    const wrapper = shallow(<Login />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})