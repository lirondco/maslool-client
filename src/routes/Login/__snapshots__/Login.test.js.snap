// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login route component renders the complete page 1`] = `
<section
  className="loginpage"
>
  <LoginForm
    onLoginSuccess={[Function]}
  />
</section>
`;
